packages/extension-ui/src/Popup/Signing/Request.tsx

```diff
@@ -61,24 +71,70 @@ export default function Request ({ account: { isExternal }, buttonText, isFirst,
     }
   }, [request]);
 
-  const _onCancel = (): Promise<void> =>
-    cancelSignRequest(signId)
-      .then(() => onAction())
-      .catch((error: Error) => console.error(error));
-  const _onSign = (password: string): Promise<void> =>
-    approveSignPassword(signId, password)
-      .then(() => onAction())
-      .catch((error: Error): void => {
-        setError(error.message);
-        console.error(error);
-      });
-  const _onSignature = ({ signature }: { signature: string }): Promise<void> =>
-    approveSignSignature(signId, signature)
+  const _onCancel = useCallback(
+    (): Promise<void> => cancelSignRequest(signId)
       .then(() => onAction())
-      .catch((error: Error): void => {
-        setError(error.message);
-        console.error(error);
-      });
+      .catch((error: Error) => console.error(error)),
+    [onAction, signId]
+  );
+
+  const _onSign = useCallback(
+    (password: string): Promise<void> => {
+      setIsBusy(true);
+
+      return approveSignPassword(signId, password, !!(password && isSavedPass))
+        .then((): void => {
+          setIsBusy(false);
+          onAction();
+        })
+        .catch((error: Error): void => {
+          setIsBusy(false);
+          setError(error.message);
+          console.error(error);
+        });
+    },
+    [onAction, isSavedPass, signId]
+  );
+
+  const _onSignQuick = useCallback(
+    () => _onSign(''),
+    [_onSign]
+  );
+
+  const _onSignature = useCallback(
+    ({ signature }: { signature: string }): Promise<void> =>
+      approveSignSignature(signId, signature)
+        .then(() => onAction())
+        .catch((error: Error): void => {
+          setError(error.message);
+          console.error(error);
+        }),
+    [onAction, signId]
+  );
+
+  const signButton = isLocked
+    ? (
+      <Unlock
+        buttonText={buttonText}
+        error={error}
+        isBusy={isBusy}
+        onSign={_onSign}
+      >
+        <Checkbox
+          checked={!!isSavedPass}
+          label="Don't ask me again for the next 15 minutes"
```

Comment:
Not familiar with the code of the extension (yet). Would it be possible to have this `15` not hard-coded without introducing too much complexity?

Reply:
The extension is as minimal as possible (and way too complex, just don't get time to fix...). Happy to consider it if people explicitly do ask for it. Since it means -

(a) extending settings to allow you to enter
(b) storing it with the settings
(c) making it available to both UI & background

Initially thought about a dropdown for 1m, 5m, 15m, 30m & 60m, which is less complex than the above, but way too many clicks again. (Although it could default to some sane-ish)

Reply:
Ah sorry, this is not what I meant.
I meant that this "15" itself is duplicated and harcoded in 2 different placed, although it would have been more generic to export a variable `const DELAY = 15` next to [PASSWORD_EXPIRY](https://github.com/polkadot-js/extension/pull/421/files#diff-037fe702f2f94ba52ce14b32737d9c74R24) and use this "DELAY" in the label.

I'm totally with you on the 15 as a good starting point, a dropdown or anythibg else would be too heavy visually.

Reply:
I vote for multiple options. If you don't want dropdowns, then what about chips? Like:
[ 15 mins ] [ 30 mins ] [ 60 mins ]